src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -18,38 +18,100 @@
+            HttpRequest.sendGETRequestWithResponseHeaders(main, "", Config.getBaseConfig(main).getOAuthProviderPublicServiceUrl() + "/oauth2/auth", queryParamsForHydra, 10000, 10000, null, responseHeaders);
+
+            if(!responseHeaders.isEmpty() && responseHeaders.containsKey(LOCATION_HEADER_NAME)) {
+                String locationHeaderValue = responseHeaders.get(LOCATION_HEADER_NAME);
+
+                if (locationHeaderValue.contains(publicOAuthProviderServiceUrl)){
+                    String error = getValueOfQueryParam(locationHeaderValue, ERROR_LITERAL);
+                    String errorDescription = getValueOfQueryParam(locationHeaderValue, ERROR_DESCRIPTION_LITERAL);
+                    throw new OAuthAuthException(error, errorDescription);
+                }
+
+                if (locationHeaderValue.contains("localhost:3000") || locationHeaderValue.contains("127.0.0.1:3000")) {
+                    redirectTo = locationHeaderValue.replace("localhost:3000", "{apiDomain}").replace("127.0.0.1:3000", "{apiDomain}");
+                } else {
+                    redirectTo = locationHeaderValue;
+                }
+            }
+            if(responseHeaders.containsKey(COOKIES_HEADER_NAME)){
+                String allCookies = responseHeaders.get(COOKIES_HEADER_NAME);
+
+                cookies = Collections.singletonList(allCookies);
+            }
         }
 
-        // TODO: parse url resposne and send appropriate reply from this API.
+        return new OAuthAuthResponse(redirectTo, cookies);
+    }
+
+    private static Map<String, String> constructHydraRequestParamsForAuthorizationGETAPICall(String clientId,
+                                                                                             String redirectURI, String responseType, String scope, String state) {
+        Map<String, String> queryParamsForHydra = new HashMap<>();
+        queryParamsForHydra.put("client_id", clientId);
+        queryParamsForHydra.put("redirect_uri", redirectURI);
+        queryParamsForHydra.put("scope", scope);
+        queryParamsForHydra.put("response_type", responseType);
+        queryParamsForHydra.put("state", state);
+        return  queryParamsForHydra;
+    }
+
+    private static String getValueOfQueryParam(String url, String queryParam){
+        String valueOfQueryParam = "";
+        if(!queryParam.endsWith("=")){
+            queryParam = queryParam + "=";
+        }
+        int startIndex = url.indexOf(queryParam) + queryParam.length(); // start after the '=' sign
+        int endIndex = url.indexOf("&", startIndex);
+        if (endIndex == -1){
+            endIndex = url.length();
+        }
+        valueOfQueryParam = url.substring(startIndex, endIndex); // substring the url from the '=' to the next '&' or to the end of the url if there are no more &s
+        return URLDecoder.decode(valueOfQueryParam, StandardCharsets.UTF_8);
     }
```

Comment:
Maybe instead of manually parsing it, we should use some lib that auto parses a url and returns a query param? This is fine too though, as long as it's extensively tested.

Reply:
yeah, using a full lib for only this seemd a bit overkill, but I will consider it.